### PR TITLE
♻️ Split partial application and curry

### PR DESCRIPTION
**Pull request by BrendanFDMoore**
✨ Add point-free
✨ Add simple compose

### CODE REVIEW

File: deck.mdx [hunk 1]
```diff
@@ -130,25 +130,99 @@ What are they?
...
+- 2-ary: a function with *two* arguments
+- N-ary: a function with _n_ arguments
+---
+What other specific arity are we familiar with?
+---
+## Ternary operator `? :`
+Three arguments: a condition and two functions
+`(condition, f, g) => if (condition) { f } else { g }`
+---
+# Partial Application
```
Comment (stefannew) [inline deck.mdx]: Two headers here

File: deck.mdx [hunk 2]
```diff
@@ -130,25 +130,99 @@ What are they?
...
+---
+# Partial Application
+- Providing some, but not all, arguments
+- Produce a new function with some values "built in"
+- Allows us to "bind" some arguments and evaluate returned functions later
+- Resulting function has lower arity (ie, fewer arguments)
```
Comment (stefannew) [inline deck.mdx]: `i.e.`

File: deck.mdx [hunk 3]
```diff
@@ -130,25 +130,99 @@ What are they?
...
+    { lines: [6, 7], notes: 'The second function can be stored as a variable' },
+    { lines: [9, 10], notes: 'and be evaluated later' },
+    { lines: [6, 7], notes: 'Curried functions allow for "point-free" style' },
+  ]}
+/>
+---
+# Point-free
+---
+# Point-free
+Omit unecessary arguments
```
Comment (stefannew) [inline deck.mdx]: Also called `Tacit programming` - maybe mention that in case people have heard it by another name?

File: deck.mdx [hunk 4]
```diff
@@ -189,9 +263,16 @@ What are they?
...
+- Always keep an eye on the goal of readability and maintainability
+- Since javascript allows multiple paradigms, _use when appropriate_
 ---
 # Codebase Example: Selectors
 Simplifying our codebase for ease of reasoning and maintenance
 ---
-
+# Exercises 
+- Write `reduce`
+- Write `filter` and `map` using your `reduce`
```
Comment (stefannew) [inline deck.mdx]: Will you post solutions for these problems, perhaps in a following slide?

Comment (BrendanFDMoore) [reply deck.mdx]: TBD -- intent is for these to be a group exercise on computers. Not sure yet.

File: deck.mdx (hunk 3 above)
Comment (BrendanFDMoore) [reply deck.mdx]: Note added.

**Pull request merged**